tests/app/views/test_buyers.py

```diff
@@ -1874,6 +1880,394 @@ def test_404_if_lot_does_not_allow_brief(self, data_api_client):
+                    },
+                    {
+                        "evidence": "Able to identify fake hobnobs and custard cremes a mile off"
+                    },
+                    {
+                        "evidence": "Have visited the Flagstaff House Museum of Tea Ware in Hong Kong"
+                    }
+                ],
+                "dayRate": "1000",
+                "blah": [False, True]
+            }
+        ]
+
+    def teardown_method(self, method):
+        self.instance = None
+
+        super(TestDownloadBriefResponsesView, self).teardown_method(method)
+
+    def test_get_responses(self):
+        po = functools.partial(mock.patch.object, autospec=True)
+
+        brief = mock.Mock()
+
+        with po(buyers, 'get_sorted_responses_for_brief') as m:
+            result = self.instance.get_responses(brief)
+
+        assert result == m.return_value
+
+        m.assert_called_once_with(brief, self.instance.data_api_client)
+
+    def test_get_question(self):
+        po = functools.partial(mock.patch.object, autospec=True)
+
+        framework_slug = mock.Mock()
+        lot_slug = mock.Mock()
+        manifest = mock.Mock()
+
+        obj = self.content_loader.get_manifest.return_value
+        content = obj.filter.return_value
+
+        result = self.instance.get_questions(framework_slug, lot_slug, manifest)
+
+        assert result == content.get_section.return_value.questions
+
+        self.content_loader.get_manifest\
+            .assert_called_once_with(framework_slug, manifest)
+
+        obj.filter.assert_called_once_with({'lot': lot_slug})
+
+        content.get_section\
+               .assert_called_once_with('view-response-to-requirements')
+
+    def test_get_question_fails_with_empty_list(self):
+        po = functools.partial(mock.patch.object, autospec=True)
+
+        framework_slug = mock.Mock()
+        lot_slug = mock.Mock()
+        manifest = mock.Mock()
+
+        self.content_loader.get_manifest.return_value\
+                           .filter.return_value\
+                           .get_section.return_value = None
+
+        result = self.instance.get_questions(framework_slug, lot_slug, manifest)
+
+        assert result == []
+
+    def test_get_context_data(self):
+        po = functools.partial(mock.patch.object, autospec=True)
+
+        self.instance.get_responses = mock.Mock()
+
+        brief = api_stubs.brief(status='closed')
+
+        kwargs = {
+            'brief_id': mock.Mock(),
+            'framework_slug': mock.Mock(),
+            'lot_slug': mock.Mock()
+        }
+
+        expected = dict(**kwargs)
+        expected['brief'] = brief['briefs']
+        expected['responses'] = self.instance.get_responses.return_value
+
+        self.instance.data_api_client.get_brief.return_value = brief
+
+        with po(buyers, 'get_framework_and_lot') as get_framework_and_lot:
+            with po(buyers, 'is_brief_correct') as is_brief_correct:
```

Comment:
I like this approach in general. It's nice that we don't need to mock the data for this, although I'd rather we do something about the 4-level nested patches.

But one problem is that autospec isn't doing much apart from checking the number of arguments. If their order or types ever change, the test will keep passing.

So if we're doing this, I think we need at least one test to test the integration.

It also feels like we need to set the return value. It's great that mock objects are truthy, but I don't think there's a good reason to obscure the actual return type.